File: lib/model/event.dart
class Event {
  final String id;
  final String name;
  final String poster;
  final int maxParticipation;
  final String applyDeadline;
  final String location;
  final String startDate;
  final String endDate;
  final String category;
  final String description;
  final num fee;
  final String society;

  const Event({
    required this.id,
    required this.name,
    required this.poster,
    required this.maxParticipation,
    required this.applyDeadline,
    required this.location,
    required this.startDate,
    required this.endDate,
    required this.category,
    required this.description,
    required this.fee,
    required this.society,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
        id: json['id'],
        name: json['name'],
        poster: json['poster'],
        maxParticipation: json['maxParticipation'],
        applyDeadline: json['applyDeadline'],
        location: json['location'],
        startDate: json['startDate'],
        endDate: json['endDate'],
        category: json['category'],
        description: json['description'],
        fee: json['fee'],
        society: json['society']);
  }

  @override
  String toString() {
    return "(id: $id, name: $name, poster: $poster, maxParticipation: $maxParticipation, applyDeadline: $applyDeadline, location: $location, startDate: $startDate, endDate: $endDate, category: $category, description: $description, fee: $fee)";
  }
}

File: src/app/model/event.ts
export interface Event {
  id?: string;
  name: string;
  poster: string;
  maxParticipation: number;
  applyDeadline: Date;
  location: string;
  startDate: Date;
  endDate: Date;
  category: EventCategory;
  description: string;
  fee: number;
  society?: string;
  version?: number;
}

export interface EventDto {
  id?: string;
  name: string;
  poster: string;
  maxParticipation: number;
  applyDeadline: string;
  location: string;
  startDate: string;
  endDate: string;
  category: EventCategory;
  description: string;
  fee: number;
  society?: string;
  version?: number;
}

export enum EventProperty {
  Id = 'Id',
  Name = 'Name',
  Poster = 'Poster',
  MaxParticipation = 'Max Participation',
  ApplyDeadline = 'Application Deadline',
  Location = 'Location',
  StartDate = 'Start Date',
  EndDate = 'End Date',
  Category = 'Category',
  Description = 'Description',
  Fee = 'Fee',
}

export interface EventFormData {
  name: string;
  poster: string;
  maxParticipation: number;
  applyDeadline: Date;
  location: string;
  date: Date[];
  category: EventCategory;
  description: string;
  fee: number;
}

export enum EventCategory {
  Sport = 'SPORT',
  Outdoor = 'OUTDOOR',
  Indoor = 'INDOOR',
}

export enum PaymentStatus {
  Paid = 'PAID',
  Unpaid = 'UNPAID',
}

export enum EventEnrollmentStatus {
  Pending = 'PENDING',
  Enrolled = 'SUCCESS',
  Rejected = 'DECLINE',
}

export interface EventEnrollmentRecord {
  itsc: string;
  studentId: string;
  paymentStatus: PaymentStatus;
  enrolledStatus: EventEnrollmentStatus;
}

export interface UpdateEventEnrollmentRecordPayload {
  eventId: string;
  studentId: string;
  paymentStatus: PaymentStatus;
  enrolledStatus: EventEnrollmentStatus;
}

export enum EventAction {
  Create = 'CREATE',
  Update = 'UPDATE',
  Fetch = 'FETCH',
  Delete = 'DELETE',
}

export interface EventAttendance {
  studentItsc: string;
  studentName: string;
  attendanceCreatedAt: Date;
}
